--- connman-1.30/src/connman.service.in~	2016-02-10 14:38:03.000000000 +0100
+++ connman-1.30/src/connman.service.in	2016-02-10 14:40:42.766477619 +0100
@@ -3,6 +3,8 @@
 After=dbus.service network-pre.target
 Before=network.target remote-fs-pre.target
 Wants=network.target remote-fs-pre.target
+# only if not NFS mount, connman will disconnect your rootfs otherwise!
+ConditionKernelCommandLine=!root=/dev/nfs
 
 [Service]
 Type=dbus
